Issue #174: remove digitally signed

eranmes opened the pull request:
This is my take on PR #232 , based off @bifurcation 's commit and addressing mine, and Rob's, comments.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -581,10 +582,12 @@ The consistency proof between hash2 and hash is PROOF(6, D\[7]) = \[i, j, k].
-algorithms defined in {{signature_algorithms}}.
+algorithms defined in {{signature_algorithms}}. Data structure are encoded
+according to Section 4 of [RFC5246] for the purpose of signing and signature
+validation.
```
We already reference RFC5246 Section 4 in the "Data Structures" section (currently numbered 1.2).

How about changing section 1.2 to say...
`Data structures are defined **and encoded** according to the conventions laid out in Section 4 of [RFC5246].`

...and then change this "Signatures" section to reference section 1.2...
`Various data structures {{data-structures}} are signed. A log MUST use one of the signature algorithms defined in {{signature_algorithms}}.`

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -929,18 +932,16 @@ which encapsulates a `SignedCertificateTimestampDataV2` structure:
-`timestamp` is equal to the timestamp from the
-`TimestampedCertificateEntryDataV2` structure encapsulated in the
-`timestamped_entry`.
+`timestamp` is equal to the timestamp from the underlying
```
We use "corresponding" in the previous section to direct the reader to this section's definition of "sct_extensions", so I suggest you do: "s/underlying/corresponding/"

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -950,10 +951,13 @@ extension that it does not understand, it SHOULD ignore that extension.
 
-The encoding of the digitally-signed element is defined in [RFC5246].
+`algorithm` indicates the signature scheme used to compute the signature.  The
+SignatureScheme enumeration is defined in [I-D.ietf-tls-tls13].
```
Add backticks around `SignatureScheme`.

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -950,10 +951,13 @@ extension that it does not understand, it SHOULD ignore that extension.
 
-`timestamped_entry` is a `TransItem` structure that MUST be of type
-`x509_entry_v2` or `precert_entry_v2` (see {{tree_leaves}}).
+`signature` is the value of a signature computed using the appropriate signature
```
Could be shortened to "`signature` is computed using..."

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -950,10 +951,13 @@ extension that it does not understand, it SHOULD ignore that extension.
-`timestamped_entry` is a `TransItem` structure that MUST be of type
-`x509_entry_v2` or `precert_entry_v2` (see {{tree_leaves}}).
+`signature` is the value of a signature computed using the appropriate signature
+algorithm for the log indicated by `log_id`.  The input to the signature is a
```
"indicated by `log_id`" seems a bit vague.  I suggest "...using the signing algorithm declared in the log's parameters (see {{log_parameters}})."

robstradling replied:
Actually, I suggest applying "s/signing algorithm/signature algorithm/" to my previous comment and to the Log Parameters section.

eranmes replied:
Done both.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -1021,7 +1024,12 @@ timestamp of the previous update.
 
-`signature` is a signature over the encoded `tree_head` field.
+`algorithm` indicates the signature scheme used to compute the signature.  The
+SignatureScheme enumeration is defined in [I-D.ietf-tls-tls13].
```
Add backticks around `SignatureScheme`.

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -1021,7 +1024,12 @@ timestamp of the previous update.
+`algorithm` indicates the signature scheme used to compute the signature.  The
+SignatureScheme enumeration is defined in [I-D.ietf-tls-tls13].
+
+`signature` is the value of a signature over the encoded `tree_head` field,
```
I suggest: s/is the value of a signature/is computed/

I don't think it's necessary to say "encoded".

eranmes replied:
Done.

eranmes commented:
Addressed all comments - I'll think on how to fold the SignatureScheme into the log parameters, since it does not make sense to duplicate this information.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -681,7 +682,8 @@ Hash Algorithm:
 : The hash algorithm used for the Merkle Tree (see {{hash_algorithms}}).
 
 Signing Algorithm:
```
Nit: s/Signing/Signature/

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -1021,7 +1020,9 @@ timestamp of the previous update.
 `tree_head` contains the latest tree head information (see {{tree_head}}).
 
-`signature` is a signature over the encoded `tree_head` field.
+`signature` is computed over the encoded `tree_head` field,
```
I think it's unnecessary to say "encoded".

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -1021,7 +1020,9 @@ timestamp of the previous update.
 
-`signature` is a signature over the encoded `tree_head` field.
+`signature` is computed over the encoded `tree_head` field,
+computed using the appropriate signature algorithm for the log indicated by
```
Repetition of "computed".

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -1021,7 +1020,9 @@ timestamp of the previous update.
-`signature` is a signature over the encoded `tree_head` field.
+`signature` is computed over the encoded `tree_head` field,
+computed using the appropriate signature algorithm for the log indicated by
+`algorithm`.
```
"algorithm" isn't there any more.

I suggest rewriting this paragraph as:
`signature` is computed over the `tree_head` field using the signature algorithm declared in the log's parameters (see {{log_parameters}}).

eranmes replied:
Done.

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -950,10 +949,12 @@ extension that it does not understand, it SHOULD ignore that extension.
-`x509_entry_v2` or `precert_entry_v2` (see {{tree_leaves}}).
+The input to the signature is a `TransItem` structure
+that MUST be of type `x509_entry_v2` or `precert_entry_v2` (see
+{{tree_leaves}}).
```
I suggest combining this paragraph and the previous paragraph as follows:

`signature` is computed over a `TransItem` structure of type `x509_entry_v2` or `precert_entry_v2` (see {{tree_leaves}}) using the signature algorithm declared in the log's parameters (see {{log_parameters}}).

eranmes replied:
Done.

eranmes commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -681,7 +682,8 @@ Hash Algorithm:
 Signing Algorithm:
-: The signing algorithm used (see {{signatures}}).
+: The signature algorithm used (see {{signatures}}), which SHOULD be a value
+  from the `SignatureScheme` enumeration, defined in [I-D.ietf-tls-tls13].
```
Done.

eranmes commented:
Addressed all comments, PTAL.
Are you suggesting removing the reference to "SignatureScheme value" from the table at the end?

robstradling commented:
No, it's the "Value" column in our signature algorithms table that I think is redundant and can now be removed.

I suggest:
* moving the "SignatureScheme value" column to be the leftmost column in the table.
* duplicating the actual values from the TLS 1.3 registry.  (e.g., write "ecdsa_secp256r1_sha256(0x0403)" instead of just "ecdsa_secp256r1_sha256").
* ditching our "Experimental Use" row.
* writing "private_use(0xFE00..0xFFFF)" for our "Private Use" row.
* ditching our "Unassigned" row, because our signature algorithm registry isn't really going to be assigning.  (Rather, it's basically just profiling the TLS SignatureScheme registry).

robstradling commented on draft-ietf-trans-rfc6962-bis.md:
```diff
@@ -680,8 +681,10 @@ Base URL:
-: The signing algorithm used (see {{signatures}}).
+Signature Algorithm:
+: The signature algorithm used (see {{signatures}}), which SHOULD be a value
+  from the TLS SignatureScheme Registry, that IANA is asked to establish in
```
Nit: Delete the comma before "that".

eranmes replied:
Done.

eranmes commented:
@robstradling PTAL.
It's unclear to me if we're requesting IANA to track TLS 1.3 and only allow values into the registry in 6962-bis if they've been added to the TLS 1.3 SignatureScheme registry, either way it seems to me a reasonable approach.

robstradling commented:
Perhaps it would make sense to move this text from the "Log Parameters" section...
`, which SHOULD be a value
  from the TLS SignatureScheme Registry that IANA is asked to establish in
  [I-D.ietf-tls-tls13].`
...into the "Expert Review guidelines" section?

The "Log Parameters" section already references the "Signatures" section, which says that logs MUST use a signature algorithm that's in our registry.  Log operators don't need to directly consider the "SHOULD be a value from the TLS SignatureScheme registry" requirement, but Expert Reviewers certainly do.

eranmes commented:
Done.
